.github/workflows/test_on_push.yml

```
@@ -63,3 +64,29 @@ jobs:
         uses: codecov/codecov-action@v4.5.0
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
+
+  run_doctests:
+    needs: style
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    name: Doctests (ubuntu-latest / Python 3.11)
```

```suggestion
    name: Doctests (ubuntu-latest / Python 3.12)
```

Maybe we can test with Python 3.12 – shouldn't be a problem?